controller/src/main/java/io/pravega/controller/store/stream/InMemoryStream.java

```diff
@@ -170,12 +170,12 @@ private void handleStreamMetadataExists(final long timestamp, CompletableFuture<
     }
 
     @Override
-    CompletableFuture<StreamConfiguration> getConfigurationData() {
+    CompletableFuture<Data<Integer>> getConfigurationData() {
         synchronized (lock) {
             if (this.configuration == null) {
                 return FutureHelpers.failedFuture(StoreException.create(StoreException.Type.DATA_NOT_FOUND, getName()));
             }
-            return CompletableFuture.completedFuture(this.configuration);
+            return CompletableFuture.completedFuture(new Data<>(SerializationUtils.serialize(this.configuration), this.configuration.getVersion()));
```

Comment:
What's the advantage of using this `Data` type compared to using simply a byte array?

Reply:
Data has binary data and version (in the store). This allows us to perform CAS on the store. 

PersistentStreamBase hides the store implementation details from rest of controller's business logic. 
Based on the store implementation (we have zk and inMemory) the Data has a version returned by the store.